CONTRIBUTING.md

```diff
@@ -1,98 +1,270 @@
+### Test File Organization
 
-- Commit messages must include a "type" as described on Conventional Commits
-- Commit messages **must** start with a capital letter
-- Commit messages **must not** end with a period `.`
+Tests should be organized to mirror the source code structure:
+
+- For a source file at `/src/index.mjs`, create a test file at `/src/test/index.test.mjs`
+- For a source file at `/src/utils/parser.mjs`, create a test file at `/src/utils/test/parser.test.mjs`
+- Test files should use the `.test.mjs` extension
+
+- For a fixture used in `/src/test/some.test.mjs`, place the fixture at `/src/test/fixture/some-fixture.mjs`.
+- When fixtures are used in multiple tests, place them in the test directory of the closest shared ancestor. For instance, if a fixture is used by both `/src/test/some.test.mjs`, and `/src/utils/test/parser.test.mjs`, the fixture belongs in `/src/test/fixtures/`.
+
+### Writing Test Code
+
+Tests should follow these guidelines:
+
+#### Use Node.js Built-in Test Runner
+
+```javascript
+import assert from 'node:assert/strict';
+import { describe, it } from 'node:test';
+```
+
+#### Test Structure
+
+Use `describe` and `it` syntax for organizing tests:
+
+```javascript
+describe('MyModule', () => {
+  describe('myFunction', () => {
+    it('should return expected result for valid input', () => {
+      // Test implementation
+      assert.strictEqual(actual, expected);
+    });
+
+    it('should throw error for invalid input', () => {
+      assert.throws(() => {
+        // Code that should throw
+      });
+    });
+  });
+});
+```
+
+#### Best Practices
+
+- **Use strict assertions**: Always use `node:assert/strict` over `node:assert`.
+- **Focused testing**: Tests should ideally only test the specific file they are intended for
+- **Use mocking**: Mock external dependencies to isolate the code under test
+- **Code splitting**: Encourage breaking down complex functionality for easier testing
+
+#### Example Test File
+
+```javascript
+// tests/index.test.mjs
+import assert from 'node:assert/strict';
+import { describe, it } from 'node:test';
+
+import { myFunction } from '../index.mjs';
+
+describe('index.mjs', () => {
+  describe('myFunction', () => {
+    it('should process valid input correctly', () => {
+      const input = 'test input';
+      const result = myFunction(input);
+      assert.strictEqual(result, 'expected output');
+    });
+
+    it('should handle edge cases', () => {
+      assert.strictEqual(myFunction(''), '');
+      assert.strictEqual(myFunction(null), null);
+    });
+
+    it('should throw for invalid input', () => {
+      assert.throws(() => myFunction(undefined), {
+        name: 'TypeError',
+        message: 'Input cannot be undefined',
+      });
+    });
+  });
+});
+```
+
+### Running Tests
+
+```bash
+# Run all tests
+node --run test
+
+# Run tests with coverage
+node --run test:coverage
+
+# Run specific test file
+node --test src/test/index.test.mjs
+```
+
+## Code Quality
```

Comment:
OOC, why are you uppercasing the first letter of each word?

Reply:
Most of the headers in the original file were cased like that

Reply:
Hmmmm, noted, maybe that was even myself, idk, but we should try to follow English norms here.